routes/user.playlist.js

```diff
@@ -205,87 +208,129 @@ router.get('/:playlist_name/export2', function(req, res) {
     })
 })
 
-router.get('/:playlist_name/export3', function(req, res) {
-    var playlistName = req.params.playlist_name
-    var songs2d = []
-    var temp = []
-    Playlist.findOne({
-        owner: req.user._id,
-        name: playlistName
-    }, function(err, playlist) {
-        if (err) return handleError(err)
-        ExportSong.find({
-                owner: playlist._id
-            })
-            .populate('song translations')
-            .exec(function(err, songs) {
-                // console.log(songs)
-                if (err) return handleError(err)
-                songs.forEach((s, i, arr) => {
-                    temp = []
-                    temp.push(s.song)
-                    s.translations.forEach((t) => {
+router.route('/:playlist_name/export3')
+    .all(function(req, res, next) {
+        playlistName = req.params.playlist_name
+        songs2d = []
+        var temp = []
+        Playlist.findOne({
+            owner: req.user._id,
+            name: playlistName
+        }, function(err, playlist) {
+            if (err) return handleError(err)
+            ExportSong.find({
+                    owner: playlist._id
+                })
+                .populate('song translations')
+                .exec(function(err, songs) {
+                    // console.log(songs)
+                    if (err) return handleError(err)
+                    songs.forEach((s, i, arr) => {
+                        temp = []
+                        temp.push(s.song)
+                        s.translations.forEach((t) => {
                             temp.push(t)
-                            // console.log(t.lyric)
+                                // console.log(t.lyric)
                         })
-                        // console.log(temp)
-                    songs2d.push(temp)
-                    if (i === arr.length - 1) {
-                        res.render('export3', {
-                            songs2d: songs2d,
-                            minLine: _.min(songs2d.map((songs) => _.min(songs.map((s) => s.lyric.length))))
-                        })
-                    }
+                        temp.push(_.min(temp.map((s) => s.lyric.length)))
```

Comment:
what's the purpose of this min? Maybe add a comment explaining what's happening here.